### PR TITLE
#3863 added attachment handling to socketio channel

### LINKED ISSUE #3863: socketio channel should also support attachments
**Description of Problem**:
Socketio channel cannot handle attachments. Because of this, attachments are handled as text messages, which cannot be parsed by Webchat JS clients.

**Overview of the Solution**:
add send_attachment method to socketio.py, like so: 

```
async def send_attachment(
        self, recipient_id: Text, attachment: Iterable[Dict[Text, Any]], **kwargs: Any
    ) -> None:
        """Sends attachment to the output."""        
        await self._send_message(self.sid, {"attachment" : attachment})
```

I can create a pull request if needed. 

**Definition of Done**:
- [ ] Tested with webchat JS clients
- [ ] Feature is documented

### PR DESCRIPTION
**Pull request by mechko**
**Proposed changes**:
- added function to handle attachment sending in socketio channel, so that Webchat JS clients can handle those attachments, see feature request #3863 
- fixes #3863 
**Status (please check what you already did)**:
- [x] made PR ready for code review
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa_nlu#code-style) for instructions)

I executed the pytest tests locally (running _python3 -m pytest tests_), there is at least no increased rate of errors, warnings or failed tests (47 failed, 823 passed, 135 warnings, 25 error). Maybe you could add some documentation on how to prepare and execute tests, and I don't know if this failed rate is the expected result (disclaimer: I have only very few knowledge about the python ecosystem). I had a look at _tests/core/test_channels.py_ and found no tests on single use cases for socketio, so I decided to not begin with this yet.

### CODE REVIEW

**Comment by akelad** [comment]
Thanks for submitting this PR, we'll give it a review as soon as possible

File: rasa/core/channels/socketio.py [hunk 1]
```diff
@@ -98,6 +98,12 @@ def __init__(self, sio, sid, bot_message_evt):
 
         await self.sio.emit(self.bot_message_evt, **json_message)
 
+    async def send_attachment(
+        self, recipient_id: Text, attachment: Iterable[Dict[Text, Any]], **kwargs: Any
```
Comment (wochinge) [inline rasa/core/channels/socketio.py]: Why `Iterable`? Should be a `Dict`, right?
```suggestion
        self, recipient_id: Text, attachment:Dict[Text, Any], **kwargs: Any
```

File: rasa/core/channels/socketio.py [hunk 2]
```diff
@@ -98,6 +98,12 @@ def __init__(self, sio, sid, bot_message_evt):
...
+    ) -> None:
+        """Sends attachment to the output."""
```
Comment (wochinge) [inline rasa/core/channels/socketio.py]:
```suggestion
        """Sends an attachment to the user."""
```

**Comment by wochinge** [comment]
And the thing with the test coverage is ok :+1:

**Comment by mechko** [comment]
Sorry, I‘m on vacation currently, I can handle this earliest by the end of July.

**Comment by wochinge** [comment]
@mechko No problem :+1:

**Comment by wochinge** [comment]
@mechko How was your vacation? 🌴

**Comment by mechko** [comment]
Sorry guys, was very busy with other topics. But thanks for merging :)

**Pull request merged**